Lightyear/Core.idr

```diff
@@ -153,4 +153,13 @@ satisfy : (Monad m, Stream tok str)
                    => (tok -> Bool)
                    -> ParserT str m tok
 satisfy p = satisfyMaybe (\t => if p t then Just t else Nothing)
+
+||| Succeed if and only if the argument parser fails.
+notFollowedBy : ParserT str m tok -> ParserT str m ()
```

Comment:
It occurs to me that `expectFailure` could sound like it’s merely to optimize the failure path or something, which something like `requireFailure` wouldn’t.

*Sigh*. Naming is hard.